app/controllers/users/access_token_controller.rb

```diff
@@ -0,0 +1,70 @@
+module Users
+  class AccessTokenController < ApplicationController
+    # Structs that help mimic auth_payload compatible with Authentication::Authenticator
+    Info = Struct.new(:name, :email, :image)
+    Extra = Struct.new(:raw_info)
+    RawInfo = Struct.new(:id, :created_at, :auth_time)
+    Payload = Struct.new(:provider, :info, :extra)
+
+    def facebook
+      # Create a payload compatible with the Authentication::Authenticator service
+      auth_payload = Payload.new(
+        "facebook",
+        Info.new(
```

Comment:
Just in case you want to use keyword arguments, they `keyword_init` argument for `Struct.new` allows that:

> If the optional `keyword_init` keyword argument is set to `true`, `.new` takes keyword arguments instead of normal arguments.

```ruby
Info = Struct.new(:name, :email, :image, keyword_init: true)
Info.new(name: "Sloan", email: "sloan@forem.com")
#<Struct:Info:0x00027d58
  email = "sloan@forem.com",
  image = nil,
  name = "Sloan"
>
```